### DISCUSSION TITLE
I want to know the right example for get_db.

### DISCUSSION THREAD

**Discussion by br-kim** [fastapi/fastapi#6519] [Questions] [answered]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's get_db examples in document.

```Python
# in SQL (Relational) Databases
def get_db():
    db = SessionLocal()
    try:
        yield db
    finally:
        db.close()

# in Testing a Database
def override_get_db():
    try:
        db = TestingSessionLocal()
        yield db
    finally:
        db.close()
```

Which example is correct?

In the first example, can catch the error at yield db and close db.

The second example catches the error when creating SessionLocal and closes the undefined db.

I think the first example is correct. Because in the second example, close the undefined db.
But is there a difference because it is a testing environment?

**Comment by ComicShrimp** [comment]
In my project I use both, and there is no difference.
I think that both ways are correct, because there is no `catch` in the `try catch`, so the exception should propagate the same way in both examples.
I don't think that are any meaningful difference other than styling, as far as a know.

**Comment by br-kim** [comment]
@ComicShrimp I understood. Thank you for your answer.

**Comment by YuriiMotov** [comment, marked as answer]
I think the first is better.
As you said, if you have any error during the creating the session, you don't have session and the attempt to close it in `finally` block will cause an exception. So, the second example is incorrect.

The best way is to use context managers as it's shown in current version of docs: https://fastapi.tiangolo.com/tutorial/sql-databases/#create-a-session-dependency
